src/Testcontainers.Consul/ConsulContainer.cs

```diff
@@ -0,0 +1,25 @@
+namespace Testcontainers.Consul;
+
+/// <inheritdoc cref="DockerContainer" />
+[PublicAPI]
+public sealed class ConsulContainer : DockerContainer
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ConsulContainer" /> class.
+    /// </summary>
+    /// <param name="configuration">The container configuration.</param>
+    /// <param name="logger">The logger.</param>
+    public ConsulContainer(ConsulConfiguration configuration, ILogger logger)
+        : base(configuration, logger)
+    {
+    }
+
+    /// <summary>
+    /// Gets the Consul connection string.
+    /// </summary>
+    /// <returns>The Consul connection string.</returns>
+    public string GetConnectionString()
+    {
+        return new UriBuilder("http", Hostname, GetMappedPublicPort(ConsulBuilder.ConsulPort)).Uri.Authority;
```

Comment:
```suggestion
        return new UriBuilder(Uri.UriSchemeHttp, Hostname, GetMappedPublicPort(ConsulBuilder.ConsulPort)).ToString();
```

Then you can simply use the connection string in the test (you do not need to prepend `http://`).